### PR TITLE
Added support for encoded responses

### PR DESCRIPTION
**Pull request by umbreak**
Fixes https://github.com/BlueBrain/nexus/issues/2588
Fixes https://github.com/BlueBrain/nexus/issues/2589

### CODE REVIEW

File: delta/plugins/storage/src/main/scala/ch/epfl/bluebrain/nexus/delta/plugins/storage/storages/routes/StoragesRoutes.scala [hunk 1]
```diff
@@ -81,7 +81,7 @@ final class StoragesRoutes(
 
   @SuppressWarnings(Array("OptionGet"))
   def routes: Route =
-    (baseUriPrefix(baseUri.prefix) & replaceUri("storages", schemas.storage, projects)) {
+    (baseUriPrefix(baseUri.prefix) & replaceUri("storages", schemas.storage, projects) & encodeResponse) {
```
Comment (imsdu) [inline delta/plugins/storage/src/main/scala/ch/epfl/bluebrain/nexus/delta/plugins/storage/storages/routes/StoragesRoutes.scala:84]: We can't put it only in the Main where we handle cors/exceptions ?

**Review by imsdu — CHANGES_REQUESTED** [review]

**Comment by imsdu** [comment]
* Documentation is missing
* We could create an issue to also handle compression in the storage app ?

**Comment by umbreak** [comment]
What documentation? This is part of the standard HTTP protocol, I don't feel we have to explicitly document it. Elasticsearch does not have any documentation for it either.

File: delta/plugins/blazegraph/src/main/scala/ch/epfl/bluebrain/nexus/delta/plugins/blazegraph/routes/BlazegraphViewsRoutes.scala [hunk 2]
```diff
@@ -75,7 +75,7 @@ class BlazegraphViewsRoutes(
     JsonLdEncoder.computeFromCirce(ContextValue(contexts.statistics))
 
   def routes: Route =
-    (baseUriPrefix(baseUri.prefix) & replaceUri("views", schema.iri, projects)) {
+    baseUriPrefix(baseUri.prefix) & replaceUri("views", schema.iri, projects) {
```
Comment (bogdanromanx) [inline delta/plugins/blazegraph/src/main/scala/ch/epfl/bluebrain/nexus/delta/plugins/blazegraph/routes/BlazegraphViewsRoutes.scala:78]: this kind of change yields compilation errors.

**Review by bogdanromanx — CHANGES_REQUESTED** [review]
I think it's fine if we don't include this in the docs and the addition is good, but there are some issues with the PR checks (compilation, static analysis and tests).

Was this tested manually?

File: delta/plugins/blazegraph/src/main/scala/ch/epfl/bluebrain/nexus/delta/plugins/blazegraph/routes/BlazegraphViewsRoutes.scala (hunk 2 above)
Comment (umbreak) [reply delta/plugins/blazegraph/src/main/scala/ch/epfl/bluebrain/nexus/delta/plugins/blazegraph/routes/BlazegraphViewsRoutes.scala]: I've fixed it

**Comment by umbreak** [comment]
Yes, it was tested manually. I checked with wireshark that now the interaction with ES returns gzipped content. Same for the interactions with our API.

The few errors we had on the CI were due to the last modifications I did, where I removed the `()`

**Comment by imsdu** [comment]
In documentation, we could just have a quick mention that we support it at the top of the general api reference page.
And not forget to add it later in the release notes

**Review by imsdu — APPROVED** [review]

**Review by bogdanromanx — APPROVED** [review]

**Pull request merged**